uefi/src/portable_acpi.rs

```diff
@@ -0,0 +1,347 @@
+use {
+    acpi::{
+        platform::{
+            address::GenericAddress,
+            interrupt::{
+                Apic, InterruptSourceOverride, IoApic, LocalInterruptLine, NmiLine, NmiProcessor,
+                NmiSource, Polarity, TriggerMode,
+            },
+            PlatformInfo, PmTimer, Processor, ProcessorInfo,
+        },
+        sdt::Signature,
+        AcpiHandler, AcpiTables, AmlTable,
+        InterruptModel::{self, *},
+        PowerProfile, Sdt,
+    },
+    alloc::vec::Vec,
+};
+
+/// Drop-in replacement for `acpi::platform::interrupt::Apic` that uses slices instead of vectors
+/// for easy transportation across the UEFI Boot Services boundary
```

Comment:
Slices don't have a stable ABI either. Also as there is no `#[repr(C)]` the compiler is free to reorder fields.